src/features/expandos/redditGallery.ts

```diff
@@ -1,87 +1,51 @@
-};
-
-function isImageMetadata(item: unknown): item is ImageMetadata {
-    return item instanceof Object && "u" in item && typeof item.u === "string";
-}
+// type ImageMetadata = {
+//     // y: number;
+//     // x: number;
+//     u: string;
+// };
 
 type MediaMetadataItem = {
     // status: string,
     // e: string,
     m: string;
-    p: unknown[]; //ImageMetadata[];
+    // p: unknown[]; //ImageMetadata[];
     // s: ImageMetadata,
     id: string;
 };
 
-function isMediaMetadataItem(item: unknown): item is MediaMetadataItem {
-    return (
-        item instanceof Object &&
-        "m" in item &&
-        typeof item.m === "string" &&
-        "p" in item &&
-        Array.isArray(item.p) &&
-        "id" in item &&
-        typeof item.id === "string"
-    );
-}
-
 export default class RedditGallery implements ExpandoProvider {
     sitename = "Reddit Gallery";
     urlregex = new RegExp(/https:\/\/www\.reddit\.com\/gallery\/.{7}/);
     async createGalleryData(post: HTMLDivElement) {
-        let imgMap = OrderedMap<string, string>();
-        const unsortedImgMap = new Map<string, Array<string>>();
-
-        const commentsLink = post.querySelector<HTMLAnchorElement>(".comments");
-        if (!commentsLink) {
-            console.error("Couldn't find comments link!");
-            return imgMap;
-        }
-
-        const postUrl = commentsLink.href + ".json";
-        console.log("postUrl json url", postUrl);
-        const response = await fetch(postUrl, {
+        const postLink = post.dataset.permalink;
+        const jsonUrl = `https://old.reddit.com${postLink}.json`;
+        console.log(jsonUrl);
+        const response = await fetch(jsonUrl, {
             referrerPolicy: "no-referrer",
         });
-        let postData = await response.json();
-        if (Array.isArray(postData)) {
-            postData = postData[0];
+        if (!response.ok) {
+            console.log(response);
+            throw new Error("Error in fetching gallery json data");
         }
-        console.log(postData);
-        postData = postData.data.children[0].data
-        for (const imgData of Object.values(
-            postData.media_metadata
-        )) {
-            if (!isMediaMetadataItem(imgData)) continue;
-            const lastImageMetadata = imgData.p.at(-1);
-            if (!isImageMetadata(lastImageMetadata)) continue;
+        const responseJson = await response.json();
+        const postData = responseJson[0].data.children[0].data;
+        console.debug("postData is", postData);
+
+        const galleryItems: { media_id: string; caption: string }[] =
+            postData.gallery_data.items;
+        const mediaMetadata = postData.media_metadata;
+        return galleryItems.map(({ media_id, caption }) => {
+            return {
+                imageSrc: this.getImageSrc(mediaMetadata[media_id]),
+                caption,
+            };
+        });
+    }
 
-            const link = document.createElement("a");
```

Comment:
I removed the link to the original image in the caption below the image because you can access the original image by pressing the download button instead

Reply:
Makes sense.